### PR TITLE
ci: add 'releasing/supported_bases.txt'

### PR DESCRIPTION
**Pull request by umarcor**
This is one additional step towards #622. I hoped to merge that one already, but I/we are not ready yet.

In this PR, the subdirs and `.keepme` files are removed from `releasing`. Instead, a `supported_bases.text` file is added, where the target distributions/environments (container images) are listed.

At the same time, script `docker-generate.sh` is merged into `docker-run.sh`. Instead of generating multiple dockerfiles, the one for the target task is generated and used only.

The purpose is to not build Verible in a dockerfile, which is unidiomatic. Instead, create a container image with all the dependencies, but then build Verible in a temporary container. For that reason, script `build.sh` is created. In `docker-run.sh`, the dockerfile is generated and the image is built. Then, a container is started and the build script is executed inside.

I could successfully move the execution of `github-pages-setup.sh` and `github-releases-setup.sh` into the new `build.sh`. However, moving `bazel build` produces an error on centos7, due to `BAZEL_CXXOPTS`. I could not find the problem there yet.

I propose to have this merged now, and I will then rebase #622, which will make it easier to analyze the centos7 issue.

### CODE REVIEW

File: .github/workflows/generate_matrix.sh [hunk 1]
```diff
@@ -16,8 +16,9 @@
 set -e
```
Comment (mithro) [inline .github/workflows/generate_matrix.sh:16]: Should we replace this shell script with a Python script?

File: .github/workflows/verible-ci.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ on:
   pull_request:
 
 env:
+  DOCKER_BUILDKIT: 1
```
Comment (mithro) [inline .github/workflows/verible-ci.yml:8]: This might need a comment about why BUILDKIT is being used?

File: .github/workflows/verible-ci.yml [hunk 3]
```diff
@@ -116,20 +118,23 @@ jobs:
...
       with:
-        name: verible-kythe-index
         path: kythe_output/*.kzip
 
 
   Matrix:
     runs-on: ubuntu-latest
+    name: Generate Build matrix
     outputs:
       matrix: ${{ steps.generate.outputs.matrix }}
```
Comment (mithro) [inline .github/workflows/verible-ci.yml:131]:
```suggestion
      matrix: ${{ steps.generate.outputs.matrix }}

```

File: releasing/docker-run.sh [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
Comment (mithro) [inline releasing/docker-run.sh:1]: I wonder if we should replace this bash script with a Python version?

**Review by mithro — COMMENTED** [review]
I _think_ I generally like the direction this is going but I worry about making these shell scripts more complicated (shell scripts quickly become unmaintainable).  Do you think we should rewrite them in Python?

File: .github/workflows/generate_matrix.sh (hunk 1 above)
Comment (umarcor) [reply .github/workflows/generate_matrix.sh:16]: Yes.

File: .github/workflows/verible-ci.yml (hunk 2 above)
Comment (umarcor) [reply .github/workflows/verible-ci.yml:8]: There is no specific reason to do so. It's just the second generation. It provides advantages but there is no specific new feature which we are using (yet). On some platforms, buildkit is enabled by default.

File: .github/workflows/verible-ci.yml (hunk 3 above)
Comment (umarcor) [reply .github/workflows/verible-ci.yml:131]: I don't see what's the change here...

File: releasing/docker-run.sh (hunk 4 above)
Comment (umarcor) [reply releasing/docker-run.sh:1]: Yes, but not yet. This is going towards https://github.com/umarcor/verible/blob/ci/rework/releasing/docker-run.sh. I think we can change to Python when we achieve that.

File: .github/workflows/generate_matrix.sh (hunk 1 above)
Comment (umarcor) [reply .github/workflows/generate_matrix.sh:17]: Done

File: .github/workflows/verible-ci.yml (hunk 3 above)
Comment (mithro) [reply .github/workflows/verible-ci.yml:131]: Adding a newline after matrix.....

File: .github/workflows/generate-matrix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
```
Comment (mithro) [inline .github/workflows/generate-matrix:1]: Let's make this file end in `.py`.

File: releasing/docker-run.sh (hunk 4 above)
Comment (mithro) [reply releasing/docker-run.sh:1]: Okay, SGTM.

**Review by mithro — APPROVED** [review]
If you fix the script name to end in `.py` we can merge!

**Comment by mithro** [comment]
@googlebot I consent.

**Pull request merged**